AEROGEAR-8601 - Docs for custom metrics implementation (#639)
aerogear/mobile-docs: modules/ROOT/pages/_partials/sync/server-audit-logs.adoc

const auditLogger = require('@aerogear/voyager-audit')
----

> 2. Inject the auditLogger module into {sync-server}. This will enable audit logging within your application.

Reviewer: ```suggestion
2. Inject the auditLogger module into the {sync-server}. This enables audit logging within your application.
```

The link:https://github.com/aerogear/voyager-server/blob/master/examples/auditLogging[Audit Logging Example Server Guide] has an example server based off the instructions above and shows all of the steps needed to get it running.

> Alternatively, if the default audit logger does not match the requirements, you can implement your own audit logger which implements the `AuditLogger` interface.

Reviewer: ```suggestion
Alternatively, if the default audit logger does not match your requirements, you can create an audit logger that implements the `AuditLogger` interface as defined below.
```

> Definition of the `AuditLogger` interface is below:

Reviewer: ```suggestion
.Definition of the `AuditLogger` interface
```

[source,typescript]
----
export interface AuditLogger {
  logResolverCompletion(msg: string, success: boolean, obj: any, args: any, context: any, info: GraphQLResolveInfo): void
  logConflict (msg: string, serverData: any, clientData: any, obj: any, args: any, context: any, info: GraphQLResolveInfo): void
  auditLog(msg: string, obj: any, args: any, context: any, info: GraphQLResolveInfo): void
}
----

> Following example implements an `AuditLogger` and injects it to {sync-server}. This implementation redacts the arguments using a `myCustomRedactionFunction` function.

Reviewer: ```suggestion
The following example implements an `AuditLogger` and injects it into the {sync-server}. 
The example redacts the arguments using a `myCustomRedactionFunction` function.
```

      {
        audit: {
          tag: 'AUDIT',
          logType: LOG_TYPE_CONFLICT,
          msg: msg || '',
          requestId: context && context.request ? context.request.id : '',
          operationType: info.operation.operation,
          fieldName: info.fieldName,
          parentTypeName: info.parentType.name,
          path: buildPath(info.path),
          parent: obj,
          arguments: myCustomRedactionFunction(args),
          clientInfo: context && context.request && context.request.body && context.request.body.extensions && context.request.body.extensions.metrics || undefined,
          authenticated: !!(context && context.auth && context.auth.isAuthenticated()),
          userInfo: (context && context.auth && context.auth.accessToken) ? context.auth.accessToken.content : undefined,
          conflict: true,
          conflictData: {
            message: msg,
            myCustomRedactionFunction(serverData),
            myCustomRedactionFunction(clientData),
          }
        }
      }
    ));
  }
}

// ...

const voyagerConfig = {
  auditLogger
}
const server = VoyagerServer(apolloConfig, voyagerConfig)

----

= Sending Device Information in {sync-client}

> Please see <<#sync-clien-audit-logs, {sync-client} Audit Logs>> section for more details.

Reviewer: ```suggestion
See the <<#sync-clien-audit-logs, {sync-client} Audit Logs>> section for more information.
```
